public/app/features/alerting/unified/components/rules/central-state-history/CentralAlertHistoryScene.tsx

```diff
@@ -214,200 +188,69 @@ function getSceneQuery(datasource: DataSourceInformation) {
-      body:
-        // eslint-disable-next-line
-        PanelBuilders.timeseries()
-          .setTitle('Alert Events')
-          .setDescription(
-            'Each alert event represents an alert instance that changed its state at a particular point in time. The history of the data is displayed over a period of time.'
-          )
-          .setData(getSceneQuery(datasource))
-          .setColor({ mode: 'continuous-BlPu' })
-          .setCustomFieldConfig('fillOpacity', 100)
-          .setCustomFieldConfig('drawStyle', GraphDrawStyle.Bars)
-          .setCustomFieldConfig('lineInterpolation', LineInterpolation.Linear)
-          .setCustomFieldConfig('lineWidth', 1)
-          .setCustomFieldConfig('barAlignment', 0)
-          .setCustomFieldConfig('spanNulls', false)
-          .setCustomFieldConfig('insertNulls', false)
-          .setCustomFieldConfig('showPoints', VisibilityMode.Auto)
-          .setCustomFieldConfig('pointSize', 5)
-          .setCustomFieldConfig('stacking', { mode: StackingMode.None, group: 'A' })
-          .setCustomFieldConfig('gradientMode', GraphGradientMode.Hue)
-          .setCustomFieldConfig('scaleDistribution', { type: ScaleDistribution.Linear })
-          .setOption('legend', { showLegend: false, displayMode: LegendDisplayMode.Hidden })
-          .setOption('tooltip', { mode: TooltipDisplayMode.Single })
-          .setNoValue('No events found')
-          .build() as VizPanel<Options, FieldConfig>,
-    }),
+    body:
+      // eslint-disable-next-line
+      PanelBuilders.timeseries()
+        .setTitle('Alert Events')
+        .setDescription(
+          'Each alert event represents an alert instance that changed its state at a particular point in time. The history of the data is displayed over a period of time.'
+        )
+        .setData(getQueryRunnerForAlertHistoryDataSource())
+        .setColor({ mode: 'continuous-BlPu' })
+        .setCustomFieldConfig('fillOpacity', 100)
+        .setCustomFieldConfig('drawStyle', GraphDrawStyle.Bars)
+        .setCustomFieldConfig('lineInterpolation', LineInterpolation.Linear)
+        .setCustomFieldConfig('lineWidth', 1)
+        .setCustomFieldConfig('barAlignment', 0)
+        .setCustomFieldConfig('spanNulls', false)
+        .setCustomFieldConfig('insertNulls', false)
+        .setCustomFieldConfig('showPoints', VisibilityMode.Auto)
+        .setCustomFieldConfig('pointSize', 5)
+        .setCustomFieldConfig('stacking', { mode: StackingMode.None, group: 'A' })
+        .setCustomFieldConfig('gradientMode', GraphGradientMode.Hue)
+        .setCustomFieldConfig('scaleDistribution', { type: ScaleDistribution.Linear })
+        .setOption('legend', { showLegend: false, displayMode: LegendDisplayMode.Hidden })
+        .setOption('tooltip', { mode: TooltipDisplayMode.Single })
+        .setNoValue('No events found')
+        .build() as VizPanel<Options, FieldConfig>,
   });
 }
 
-export class EventsPanelScenesObject extends SceneObjectBase<
```

Comment:
With the filter variables part of the query, and SceneQueryRunner automatically depending on the variables (and re-running queries) this object is now completely pointless so removed it

Reply:
💯